### ISSUE TITLE
The accuracy of the datasets cub is different from that in the paper

### ISSUE THREAD

**Issue by zcgit001** [euwern/proxynca_pp#7]
Hi,
Thanks for sharing the code!!!
I ran an experiment on the datasets cub with your code without any changes and found that the accuracy of the datasets cub  was only around 71.0.But, in your paper, the accuracy of the datasets cub is around 72.2.I don't know why.
Then I have carefully checked your code and paper, and I found that during data preprocessing, your code is not the same as described in the paper.
![Screenshot from 2021-04-14 12-34-18](https://user-images.githubusercontent.com/50916572/114655131-24634f80-9d1e-11eb-9a97-4ed49fb4a4f8.png)
![Screenshot from 2021-04-14 12-35-00](https://user-images.githubusercontent.com/50916572/114655223-5d9bbf80-9d1e-11eb-9a20-5808a29bdd5a.png)
So,could you tell me if is the difference in data preprocessing that leads to the difference in experimental results.

**Comment by zcgit001** [comment]
Sorry, I have looked at the source code of ‘transforms.RandomResizedCrop’ and found that this function performs the data preprocessing operation in your paper.So,the data preprocessing in your paper is the same as the data preprocessing in your code.
But, I still don't know why my experimental results are different from those in your paper.

**Comment by zcgit001** [comment]
I'm sorry. I see what the problem is.When I trained with a gpu, the accuracy rate was around 71.When I trained with two gpus, the accuracy rate was around 72.2.
Thanks again for your code!

**Comment by zcgit001** [comment]
So now there is a new problem. Why is the result different when I trained with one GPU than when I trained with two GPUs.I can't found the answer to the question.

**Comment by euwern** [comment]
Thx, zhaoxiaochung001, for reporting this issue. I suspect that the cause of this issue is due to the BatchNorm layers in ResNet architecture. BatchNorm is known to be very sensitive to batch size [1]. In PyTorch, batch statistics are calculated per GPU. Given a batch size of B, if a model is trained with a single GPU, batch size remains as B; however, if you train with two GPUs, the batch size per GPU is B/2.  

 [1] https://arxiv.org/abs/1803.08494

**Comment by zcgit001** [comment]
Hi,euwern.Thank you very much for your reply.I guess that's the cause too.Thanks again for your code.
